src/loader/index-default.js

```diff
@@ -9,9 +9,20 @@ var cwd = process.cwd();
 var fsOptions = {encoding: 'utf8'};
 
 module.exports = function load(templatePath, templateSrc, options) {
-    if (typeof templatePath === 'string' && nodePath.extname(templatePath) === '.js') {
-        // assume compiled template
-        return require(templatePath);
+    if (typeof templatePath === 'string') {
+        const parsed = nodePath.parse(templatePath);
+
+        if (parsed.ext === '.js') {
+            // assume compiled template
+            return require(templatePath);
+        } else if (parsed.ext === '.marko') {
+            // Check if precompiled js file already exists.
+            const jsFilePath = nodePath.join(parsed.dir, parsed.name + '.js');
+            try {
+                fs.accessSync(jsFilePath);
+                return require(jsFilePath);
```

Comment:
You do not want to surround the `require()` with a `try...catch` since that could hide real problems in the compiled template

Reply:
Right, missed that! One sec.